airflow/api_connexion/endpoints/dag_endpoint.py

```diff
@@ -69,7 +69,7 @@ def get_dag_details(*, dag_id: str) -> APIResponse:
     return dag_detail_schema.dump(dag)
 
 
-@security.requires_access([(permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG)])
+@requires_authentication
```

Comment:
Just curious, why no method on this one? I expected:
```python
@security.requires_access_dag("GET")
```

Reply:
> I dont think this is perfect so if you think something is off here, please call it out :).

I don't think I have enough context to call out whether something is off here or not. But your explanation does make sense!

> @security.requires_access_dag("GET") would check if the user has access to all DAGs but here this is not what we want to check.

Is this what the decorator is doing for something like `get_dag(...)` above? In that case we only want one specific dag, but we're checking if the user has access to all DAGs?

> While writing this comment I am wondering if `@security.requires_access_dag("GET")` should mean "does the user have access to one DAG" instead of "does the user have access to all DAGs". Thoughts? I dont think checking whether the user has access to all DAGs in Airflow makes sense.

I _think_ this makes more sense. And is more of the common case. But surely there are cases where an action is being done on multiple dags? (but maybe not, you would know better than I).

Reply:
> Is this what the decorator is doing for something like get_dag(...) above? In that case we only want one specific dag, but we're checking if the user has access to all DAGs?

If a dag id is passed as parameter (or in the request), then the authorization check is done only against this specific DAG. This is always the case, for now, every-time `@security.requires_access_dag("GET")` is used, a DAG id is provided. This tells me that we do not necessarily need to check whether a user has access to all DAGs. Thanks, you helped me figuring it out :)

Reply:
Haha glad I could help rubber ducky :sweat_smile: :duck:

Reply:
I just pushed the changes to address it